src/dotnet/commands/dotnet-new/NewCommandShim.cs

```diff
@@ -77,15 +75,79 @@ private static ITemplateEngineHost CreateHost()
         }
 
         private static void FirstRun(IEngineEnvironmentSettings environmentSettings, IInstaller installer)
+        {
+            var templateFolders = GetTemplateFolders(environmentSettings);
+            foreach (var templateFolder in templateFolders)
+            {
+                var layoutIncludedPackages = environmentSettings.Host.FileSystem.EnumerateFiles(templateFolder, "*.nupkg", SearchOption.TopDirectoryOnly);
+                installer.InstallPackages(layoutIncludedPackages);
+            }
+        }
+
+        private static IEnumerable<string> GetTemplateFolders(IEngineEnvironmentSettings environmentSettings)
         {
             Paths paths = new Paths(environmentSettings);
-            var templatesDir = Path.Combine(paths.Global.BaseDir, "Templates");
+            List<string> templateFoldersToInstall = new List<string>();
+
+            // First grab templates from dotnet\templates\M.m folders, in ascending order, up to our version
+            var templatesRootFolder = Path.GetFullPath(Path.Combine(paths.Global.BaseDir, "..", "..", "templates"));
```

Comment:
Can we make the casing consistent here? As in Templates (capital T) versus templates (lower case t)?

Reply:
And my suggest, if the existing one already uses upper case T, for the new one to be so as well.

Reply:
We do need to use whatever is controlled by setup though for case-sensitive OSes, right?

I think the one in the sdk folder has used T since 1.1, and I also recall that someone (Lee?) had said something about this new one being lower case - likely for consistency with other sibling folders under "dotnet".

I'm not sure how much consistency here buys us anyway, considering that the "Templates" location under SDK is going away after @johnbeisner's checkin, and including that path in the search here will be dead code which can be removed.